### PR TITLE
tests/ifacestate: moved asserts-related mocking into helper

### PR DESCRIPTION
**Pull request by stolowski**
Moved asserts-related mocking code used in ifacestate tests into a helper, for reuse in hotplug tests (which live in a different suite under ifacestate/ as well).

### CODE REVIEW

File: overlord/ifacestate/export_test.go [hunk 1]
```diff
@@ -77,3 +84,113 @@ func UpperCaseConnState() map[string]connState {
 		"APP:network CORE:network": {Auto: true, Interface: "network"},
 	}
 }
+
+type AssertsMock struct {
```
[the remark this answers is not in the document]
Comment (stolowski) [reply overlord/ifacestate/export_test.go]: Right, no strong reason, the idea was to separate it from a specific suite file. Moved it back to ifacestate_test.go

File: overlord/ifacestate/export_test.go [hunk 2]
```diff
@@ -77,3 +84,113 @@ func UpperCaseConnState() map[string]connState {
...
+	Db           *asserts.Database
+	storeSigning *assertstest.StoreStack
+	brandSigning *assertstest.SigningDB
+}
+
+func (am *AssertsMock) MockAsserts(c *check.C, st *state.State) {
```
[the remark this answers is not in the document]
Comment (stolowski) [reply overlord/ifacestate/export_test.go]: Thanks, I was looking for a better name, and this is a good name, changed.

**Comment by stolowski** [comment]
Ready for re-review

File: overlord/ifacestate/ifacestate_test.go [hunk 3]
```diff
@@ -57,32 +57,135 @@ import (
...
+
+	brandPubKey, err := am.brandSigning.PublicKey("")
+	c.Assert(err, IsNil)
+	brandAccKey := assertstest.NewAccountKey(am.storeSigning, brandAcct, nil, brandPubKey, "")
+	err = assertstate.Add(st, brandAccKey)
+	c.Assert(err, IsNil)
+	st.Unlock()
+}
+
+func (am *AssertsMock) MockModel(c *C, st *state.State, extraHeaders map[string]interface{}) {
```
Comment (pedronis) [inline overlord/ifacestate/ifacestate_test.go]: you could capture the State again in SetupAsserts and avoid needing to pass it in again here

Comment (stolowski) [reply overlord/ifacestate/ifacestate_test.go]: Good idea, done. Thank you!

**Pull request merged**